packages/opentelemetry-types/src/trace/trace_state.ts

```diff
@@ -14,6 +14,43 @@
  * limitations under the License.
  */
 
+/**
+ * Tracestate carries system-specific configuration data, represented as a list
+ * of key-value pairs. TraceState allows multiple tracing systems to
+ * participate in the same trace.
+ */
 export interface TraceState {
-  // TODO
+  /**
+   * Adds or updates the TraceState that has the given `key` if it is
+   * present. The new State will always be added in the front of the
+   * list of states.
+   *
+   * @param key key of the TraceState entry.
+   * @param value value of the TraceState entry.
+   */
+  set(key: string, value: string): void;
```

Comment:
Should there be an `unset` or some kind of value that can be passed to `set` to unset?

Reply:
Makes sense, what do you think about `remove` operation? I am fine with `unset` also.

Reply:
I personally prefer usually these combinations: add/remove or set/unset.

Reply:
Added `unset` method.

Reply:
@rochdev I think I addressed all of the comments by updating the PR. PTAL